$NetBSD$

Hunk #0:
  Don't let *_STAGE0 to be empty even when we're bootstrapping from
  HC.

Hunk #1:
  Use of the notorious '==' operator of test(1).


--- configure.ac.orig	2012-02-01 18:10:32.000000000 +0000
+++ configure.ac
@@ -117,7 +117,12 @@ AC_ARG_ENABLE(bootstrap-with-devel-snaps
         EnableBootstrapWithDevelSnaphost=NO
 )
 
-if test "$WithGhc" != ""; then
+if test "$WithGhc" = ""; then
+  CC_STAGE0='$(CC)'
+  AR_STAGE0='$(AR)'
+  AR_OPTS_STAGE0='$(AR_OPTS)'
+  ArSupportsAtFile_STAGE0='$(ArSupportsAtFile)'
+else
   FPTOOLS_GHC_VERSION([GhcVersion], [GhcMajVersion], [GhcMinVersion], [GhcPatchLevel])dnl
 
   if test "$GhcMajVersion" = "unknown" -o "$GhcMinVersion" = "unknown"; then
@@ -496,7 +501,7 @@ dnl ** check for dtrace (currently only 
 HaveDtrace=NO
 AC_PATH_PROG(DtraceCmd,dtrace)
 if test -n "$DtraceCmd"; then
-  if test "x$TargetOS_CPP-$TargetVendor_CPP" == "xdarwin-apple" -o "x$TargetOS_CPP-$TargetVendor_CPP" == "xsolaris2-unknown"; then
+  if test "x$TargetOS_CPP-$TargetVendor_CPP" = "xdarwin-apple" -o "x$TargetOS_CPP-$TargetVendor_CPP" = "xsolaris2-unknown"; then
     HaveDtrace=YES
   fi
 fi
